src/libs/cluster-utils.js

```diff
@@ -78,6 +78,9 @@ export const trimClustersOldestFirst = _.flow(
 
 export const currentCluster = _.flow(trimClustersOldestFirst, _.last)
 
+export const collapsedClusterStatus = cluster => {
+  return cluster && (cluster.patchInProgress ? 'Reconfiguring' : cluster.status) // NOTE: preserves null vs undefined
```

Comment:
Since we are setting this here, could we make this 'Updating' rather than 'Reconfiguring' so we don't have to do the text transforms?

Reply:
'Updating' is already a real status that we get back when adjusting the number of workers in a spark cluster. We wanted this new state to display as 'Updating', but behave slightly differently (we need to kick you out of your notebook because of the machine stop, whereas the real 'Updating' status takes place while the machine is running)

Reply:
TOL: What if `Reconfiguring` someday becomes a real state returned from GCP/leo? I know it's probably unlikely, but I worry about this kind of thing and I think there are other ways to accomplish this that would avoid that potential problem... and possibly address Scout's comment at the same time.

(Side note: I won't be able to review this in one sitting, so I'm going to add single comments instead of queueing them up in a GitHub "review".)

Reply:
That's certainly a possibility. The two general options I see are 1. picking a value that's less likely to conflict, e.g. `'LeoReconfiguring'`, or 2. stop rolling the flag in and treat it as a separate value. I avoided 2 in the initial implementation because it would have involved much more rewriting of the existing logic. Thoughts?

Reply:
I think it's worth doing option 1. Not only will it avoid the conflict but it will make it clear that this status value is special when seeing it elsewhere in the code, such as when mixed with other values in `Utils.switchCase` clauses and the case of calculating the display value that Scout commented on.

The consequence I see of option 2 is that the uses of `Utils.switchCase` on status would have to be transformed into `Utils.cond`. I agree that would generate a larger diff, but I'm inclined to think it would be the right thing to do in the absence of using option 1. (I would be open to discussion about that, and also whether or not that could be a follow-on refactor in a separate PR).

Side note: I had an even larger change in mind involving gathering the behavior around different cluster states and putting them in one place instead of having it spread throughout the code. I think I've talked myself out of that but wanted to mention that my gut told me to explore that option in case anyone else who ponders it might reach a different conclusion.